src/main/java/bio/terra/service/dataset/flight/ingest/IngestCreateParquetFilesStep.java

```diff
@@ -87,8 +99,17 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
 
   @Override
   public StepResult undoStep(FlightContext context) {
+    FlightMap workingMap = context.getWorkingMap();
     azureSynapsePdao.dropTables(
         List.of(IngestUtils.getSynapseIngestTableName(context.getFlightId())));
+
+    AzureStorageAccountResource storageAccountResource =
+        workingMap.get(
+            CommonMapKeys.DATASET_STORAGE_ACCOUNT_RESOURCE, AzureStorageAccountResource.class);
+    String parquetFilePath =
+        FolderType.METADATA.getPath(workingMap.get(IngestMapKeys.PARQUET_FILE_PATH, String.class));
```

Comment:
I think "FolderType.METADATA.getPath()" appends "metadata/" to the file path and then it looks like AzureBlobStoragePdao.deleteBlobParquet() is appending that again.

Reply:
I added the testing strategy.

The `blobUrl` has `.../metadata/metadata/...` and the call to `blobParts.getBlobName()` in the existing `deleteBlobParquet` method strips the first `metadata/` prefix, which passes the correct path to `deleteBlobsWithPrefix`. I copied this approach from the `undoStep` in `IngestCreateScratchParquetFilesStep`, which also duplicates the path ( `.../scratch/scratch/...` ).

Perhaps @tlangs has more context on why it was originally done this way.

Reply:
Good eyes, both.  We've exposed a latent bug in how these blob URLs were constructed.  The deletion does appear to work as expected, but the lack of preexisting unit tests made it hard to be confident about that behavior.

What triggered my "aha!" moment was realizing that a valid blob URL takes this form:
`<storageAccountBaseUrl>/<containerId>/<blobName>`

And more specifically in the case of our published Parquet files:
`https://<storageAccountName>.blob.core.windows.net/<datasetId>/metadata/parquet/table/flight-<flightid>.parquet/..`

Breaking that down:
```
storageAccountBaseUrl = https://<storageAccountName>.blob.core.windows.net
containerId = <datasetId>
blobName = metadata/parquet/table/flight-<flightid>.parquet/..
```

The latent bug was incorrectly using another instance of `metadata` or `scratch` instead of the actual container name:
```
storageAccountBaseUrl = https://<storageAccountName>.blob.core.windows.net
containerId = metadata  ## incorrect, but we only consider the blobName for prefix-based deletion
blobName = metadata/parquet/table/flight-<flightid>.parquet/..
```

As long as the new unit tests verify the expected behavior -- that we are deleting blobs with the expected prefix -- then I'd be comfortable with filing a separate ticket to fix this latent bug, since it doesn't presently impact users.  Thoughts?

Reply:
I ended up fixing it in this ticket and the unit tests check the behavior :)